sql/core/src/main/scala/org/apache/spark/sql/execution/commands/parsers.scala

```diff
@@ -0,0 +1,420 @@
+              val k = unquoteString(cleanIdentifier(key.text))
+              val v = unquoteString(cleanIdentifier(value.text))
+              (k, Some(v))
+          }
+      }.toMap
+  }
+
+  override def isDefinedAt(node: ASTNode): Boolean = node.text == "TOK_ALTERTABLE"
+
+  override def apply(v1: ASTNode): LogicalPlan = v1.children match {
+    case (tabName @ Token("TOK_TABNAME", _)) :: rest =>
+      val tableIdent: TableIdentifier = base.extractTableIdent(tabName)
+      val partitionSpec = base.getClauseOption("TOK_PARTSPEC", v1.children)
+      val partition = partitionSpec.flatMap(parsePartitionSpec)
+      matchAlterTableCommands(v1, rest, tableIdent, partition)
+    case _ =>
+      throw new NotImplementedError(v1.text)
+  }
+
+  def matchAlterTableCommands(
+      node: ASTNode,
+      nodes: Seq[ASTNode],
+      tableIdent: TableIdentifier,
+      partition: Option[Map[String, Option[String]]]): LogicalPlan = nodes match {
+    case rename @ Token("TOK_ALTERTABLE_RENAME", renameArgs) :: rest =>
+      val renamedTable = base.getClause("TOK_TABNAME", renameArgs)
+      val renamedTableIdent: TableIdentifier = base.extractTableIdent(renamedTable)
+      AlterTableRename(tableIdent, renamedTableIdent)(node.source)
+
+    case Token("TOK_ALTERTABLE_PROPERTIES", args) :: rest =>
+      val setTableProperties = extractTableProps(args.head)
+      AlterTableSetProperties(
+        tableIdent,
+        setTableProperties)(node.source)
+
+    case Token("TOK_ALTERTABLE_DROPPROPERTIES", args) :: rest =>
+      val dropTableProperties = extractTableProps(args.head)
+      val allowExisting = base.getClauseOption("TOK_IFEXISTS", args)
+      AlterTableDropProperties(
+        tableIdent,
+        dropTableProperties, allowExisting.isDefined)(node.source)
+
+    case Token("TOK_ALTERTABLE_SERIALIZER", serdeArgs) :: rest =>
+      val serdeClassName = unquoteString(cleanIdentifier(serdeArgs.head.text))
+
+      val serdeProperties: Option[Map[String, Option[String]]] = Option(
+        // SET SERDE serde_classname WITH SERDEPROPERTIES
+        if (serdeArgs.tail.isEmpty) {
+          null
+        } else {
+          extractTableProps(serdeArgs.tail.head)
+        }
+      )
+
+      AlterTableSerDeProperties(
+        tableIdent,
+        Some(serdeClassName),
+        serdeProperties,
+        partition)(node.source)
+
+    case Token("TOK_ALTERTABLE_SERDEPROPERTIES", args) :: rest =>
+      val serdeProperties: Map[String, Option[String]] = extractTableProps(args.head)
+
+      AlterTableSerDeProperties(
+        tableIdent,
+        None,
+        Some(serdeProperties),
+        partition)(node.source)
+
+    case (bucketSpec @ Token("TOK_ALTERTABLE_CLUSTER_SORT", _)) :: rest =>
+      val (buckets, noClustered, noSorted) = bucketSpec match {
+        case Token("TOK_ALTERTABLE_CLUSTER_SORT", clusterAndSoryByArgs :: Nil) =>
+          clusterAndSoryByArgs match {
+              case Token("TOK_ALTERTABLE_BUCKETS", bucketArgs) =>
```

Comment:
Because that rule `tableBuckets` is used in other places, I think we better don't touch it.